pkg/services/profiles/add_test.go

```diff
@@ -0,0 +1,342 @@
+package profiles_test
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/weaveworks/weave-gitops/pkg/api/profiles"
+	"github.com/weaveworks/weave-gitops/pkg/git"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders/gitprovidersfakes"
+	"github.com/weaveworks/weave-gitops/pkg/helm"
+	"github.com/weaveworks/weave-gitops/pkg/logger/loggerfakes"
+	"github.com/weaveworks/weave-gitops/pkg/services/profiles"
+	"github.com/weaveworks/weave-gitops/pkg/vendorfakes/fakegitprovider"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/fluxcd/helm-controller/api/v2beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes/fake"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/testing"
+	"sigs.k8s.io/yaml"
+)
+
+var addOptions profiles.AddOptions
+
+var _ = Describe("Add", func() {
+	var (
+		gitProviders *gitprovidersfakes.FakeGitProvider
+		profilesSvc  *profiles.ProfilesSvc
+		clientSet    *fake.Clientset
+		fakeLogger   *loggerfakes.FakeLogger
+		fakePR       *fakegitprovider.PullRequest
+	)
+
+	BeforeEach(func() {
+		gitProviders = &gitprovidersfakes.FakeGitProvider{}
+		clientSet = fake.NewSimpleClientset()
+		fakeLogger = &loggerfakes.FakeLogger{}
+		fakePR = &fakegitprovider.PullRequest{}
+		profilesSvc = profiles.NewService(clientSet, fakeLogger)
+
+		addOptions = profiles.AddOptions{
+			ConfigRepo: "ssh://git@github.com/owner/config-repo.git",
+			Name:       "podinfo",
+			Cluster:    "prod",
+			Namespace:  "weave-system",
+			Version:    "latest",
+		}
+	})
+
+	It("creates a helm release with the latest available version of the profile", func() {
+		gitProviders.RepositoryExistsReturns(true, nil)
+		gitProviders.GetDefaultBranchReturns("main", nil)
+		gitProviders.GetRepoFilesReturns(makeTestFiles(), nil)
+		clientSet.AddProxyReactor("services", func(action testing.Action) (handled bool, ret restclient.ResponseWrapper, err error) {
+			return true, newFakeResponseWrapper(getProfilesResp), nil
+		})
+		fakePR.GetReturns(gitprovider.PullRequestInfo{
+			WebURL: "url",
+		})
+		gitProviders.CreatePullRequestReturns(fakePR, nil)
+		Expect(profilesSvc.Add(context.TODO(), gitProviders, addOptions)).Should(Succeed())
+		Expect(gitProviders.RepositoryExistsCallCount()).To(Equal(1))
```

Comment:
Are we missing more assertions around what fakes were called and with what arguments? e.g.
```
Expect(gitProviders.RepositoryExistsCallCount()).To(Equal(1)) # already here
Expect(gitProviders.RepositoryExistsArgsForCall(0)).To(Equal(<something>))

Expect(gitProviders.GetDefaultBrancnCallCount()).To(Equal(1))
Expect(gitProviders.GetDefaultBrancnArgsForCall(0)).To(Equal(<something>))



```

Reply:
also what does `latest` get converted too here?